src/formats.rs

```diff
@@ -11,8 +11,14 @@ pub enum InputFormat {
     ProgramsList
 }
 
+pub struct Input {
+    pub programs : Vec<String>, // Program strings. 
+    pub tasks : Vec<String>, // Task names for each corresponding string.
+    pub prev_dc_inv_to_inv_strs: Vec<(String, String)>, // Vec of [#Dreamcoder invention, fn_i] tuples for any existing inventions in the DSL.
+}
+
 impl InputFormat {
-    pub fn load_programs_and_tasks(&self, path: &Path) -> Result<(Vec<String>, Vec<String>, usize), String> {
+    pub fn load_programs_and_tasks(&self, path: &Path) -> Result<Input, String> {
```

Comment:
This function name has outlived its use. I suggest we replace it with just `pub fn load` or something along those lines